packaging/language/maven_artifact.py

```diff
@@ -208,13 +214,10 @@ def _request(self, url, failmsg, f):
                 "User-Agent": self.user_agent,
                 "Authorization": "Basic " + base64.b64encode(self.username + ":" + self.password)
             }
-        req = Request(url, None, headers)
-        try:
-            response = urlopen(req)
-        except HTTPError, e:
-            raise ValueError(failmsg + " because of " + str(e) + "for URL " + url)
-        except URLError, e:
-            raise ValueError(failmsg + " because of " + str(e) + "for URL " + url)
+
+        response, info = fetch_url(self.module, url, headers=headers)
```

Comment:
I looked briefly at the fetch_url function - it's not clear to me where a user would specify where the trusted ca keystore is if they are using an internal CA.

Additionally, since we're switching to fetch_url should we not also let it handle the username/password functionality?